circuits/pol/src/lib.rs

```diff
@@ -1 +1,67 @@
+use std::{fs::read_to_string, io, io::Write as _, path::PathBuf, sync::LazyLock};
 
+use circuits_utils::find_binary;
+use tempfile::NamedTempFile;
+
+const BINARY_NAME: &str = "pol";
+const BINARY_ENV_VAR: &str = "NOMOS_POL";
+
+static BINARY: LazyLock<PathBuf> = LazyLock::new(|| {
+    find_binary(BINARY_NAME, BINARY_ENV_VAR).unwrap_or_else(|error_message| {
+        panic!("Could not find the required '{BINARY_NAME}' binary: {error_message}");
+    })
+});
+
+/// Runs the `pol` circuit to generate a witness from the provided inputs.
+///
+/// # Arguments
+///
+/// * `inputs_file` - The path to the file containing the public and private
+///   inputs.
+/// * `witness_file` - The path to the file where the witness will be written.
+///
+/// # Returns
+///
+/// An `io::Result<PathBuf>` which contains the path to the witness file if
+/// successful, or an `io::Error` if the command fails.
+pub fn pol(inputs_file: &PathBuf, witness_file: &PathBuf) -> io::Result<PathBuf> {
```

Comment:
I'm wondering if we really want this. Unless wrapped, what we would want is to provide the inputs directly from code, not really a file (cc @thomaslavaur). This implementation should handle dumping into a file those inputs with the proper format.